ludwig/models/modules/sequence_decoders.py

```diff
@@ -66,26 +66,35 @@ def __init__(
         self.num_classes = num_classes
         self.max_sequence_length = max_sequence_length
 
-        if self.is_timeseries:
+        if is_timeseries:
             self.vocab_size = 1
         else:
             self.vocab_size = self.num_classes
 
         self.embeddings_dec = Embedding(num_classes, embedding_size)
         self.decoder_cell = LSTMCell(state_size)
 
-        if self.attention_mechanism == 'bahdanau':
-            pass
-        elif self.attention_mechanism == 'luong':
-            pass
-        else:
-            raise ValueError(
-                "Attention specificaiton '{}' is invalid.  Valid values are "
-                "'bahdanau' or 'luong'.".format(self.attention_mechanism))
+        if attention_mechanism:
+            if attention_mechanism == 'bahdanau':
+                pass
+            elif attention_mechanism == 'luong':
+                self.attention_mechanism = tfa.seq2seq.LuongAttention(
+                    state_size,
+                    None,  # todo tf2: confirm on need
+                    memory_sequence_length=max_sequence_length  # todo tf2: confirm inputs or output seq length
```

Comment:
this should be the length of the input not considering padding if I'm correct. At least this is how it worked in TF1. I had a function to figure out that value, `sequence_length_3D`.